docs/content/preview/yugabyte-platform/install-yugabyte-platform/install-software/kubernetes.md

```diff
@@ -140,101 +128,276 @@ You install YugabyteDB Anywhere on a Kubernetes cluster as follows:
+You can add annotations to the YugabyteDB Anywhere service to create an internal load balancer instead of a public-facing one. Since every cloud provider has different annotations for doing this, refer to the following documentation:
 
-When your Kubernetes cluster nodes are spread across multiple zones, you can use this command to explicitly place the YugabyteDB Anywhere pod on specific zones, as demonstrated by the following example:
+- For Google Cloud, see [GKE docs](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing).
+- For Azure, see [AKS docs](https://docs.microsoft.com/en-us/azure/aks/internal-lb).
+- For AWS, see [EKS docs](https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html) and [AWS Load Balancer Controller docs](https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/guide/service/annotations/#lb-scheme).
+- For other providers, see [Internal load balancer](https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer).
 
-```sh
-helm install yw-test yugabytedb/yugaware/ -n yb-platform \
---version 2.15.2 --set yugaware.storageClass=yb-storage \
---set "zoneAffinity={us-west1-a,us-west1-b}"
+For example, for a GKE cluster, you would add following lines to your values file:
+
+```yaml
+# yba-values.yaml
+yugaware:
+  service:
+    # other values…
+    annotations:
+      networking.gke.io/load-balancer-type: "Internal"
 ```
 
-### toleration
+For an EKS cluster, you would use following:
 
-Kubernetes nodes could have `taints` that repel pods from being placed on it. Only pods with a `toleration` for the same `taint` are permitted. For more information, see [Taints and Tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/).
+```yaml
+# yba-values.yaml
+yugaware:
+  service:
+    # other values…
+    annotations:
+      service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
+```
 
-For example, if some of the nodes in your Kubernetes cluster are earmarked for experimentation and have a taint `dedicated=experimental:NoSchedule`, only pods with the matching toleration will be allowed; other pods will be prevented from being placed on these nodes.
+#### Set DNS name
 
-```sh
-helm install yw-test yugabytedb/yugaware/ -n yb-platform \
---version 2.15.2 --set yugaware.storageClass=yb-storage \
---values=/tmp/overrides.yaml
+If you want to access YugabyteDB Anywhere via a domain or localhost, you need to set the `tls.hostname` field to ensure that the correct TLS and Cross-Origin Resource Sharing (CORS) settings are used, as follows:
+
+```yaml
+# yba-values.yaml
+tls:
+  hostname: "yba.example.com"
+```
+
+<!-- TODO: move things from the following section to this page -->
+<!-- https://docs.yugabyte.com/preview/yugabyte-platform/troubleshoot/install-upgrade-yp-issues/#configure-load-balancer-for-helm-charts  -->
+<!-- it was added as part of PLAT-3570 -->
```

Comment:
@iSignal do you think we should move the mentioned sections to this document as part of this PR?

Reply:
@bhavin192 : we can remove the troubleshooting section you linked to, as long as both those settings are documented here.

Reply:
@bhavin192 It's a typical doc structure pattern to add troubleshooting information to a separate section specifically dedicated to troubleshooting of various aspects of the product. 
It would be better not to move the section from Troubleshooting but instead provide a link to it.

Reply:
Structure wise you are right that there should be the troubleshooting section. The problem is those two points are more of instructions rather than actual troubleshooting, those don't talk about problems much, just say add this flag that flag etc. For now I'm not touching that section as Yatish is revamping the whole Kubernetes troubleshooting guide which takes care it.